File: Scadex.Model/Dtos/Diagram/Queries/Items/DiagramDeviceDto.cs
using Scadex.Core.Model;
using static Scadex.Model.Enums.EntityEnums;

namespace Scadex.Model.Dtos.Diagram.Queries.Items;

public class DiagramDeviceDto : IDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public double CoordinateX { get; set; }
    public double CoordinateY { get; set; }
    public double Rotation { get; set; }
    public int ZIndex { get; set; }
    public bool IsLocked { get; set; }
    public bool IsVisible { get; set; }
    public bool IsActive { get; set; }
    public Guid ComponentTemplateId { get; set; }
    /// <summary>SCADA tarafindaki kimlik; ingest bu kodla cihaz cozumler.</summary>
    public string? ExternalCode { get; set; }
    public int? DeviceStatusId { get; set; }
    public string? DeviceStatusName { get; set; }
    public DateTime? LastSeen { get; set; }
    public DiagramComponentTemplateDto Template { get; set; } = null!;
    public ICollection<DiagramPinDto> Pins { get; set; } = [];
    public ICollection<DiagramIoChannelDto> IoChannels { get; set; } = [];
}

public class DiagramComponentTemplateDto : IDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public int DeviceTypeId { get; set; }
    public double Width { get; set; }
    public double Height { get; set; }
    public string BackgroundColor { get; set; } = null!;
    public string? BackgroundImageUrl { get; set; }
}


public class DiagramPinDto : IDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = null!;
    public double RelativeX { get; set; }
    public double RelativeY { get; set; }
    public HandleSide Side { get; set; }
    public PinFunction Function { get; set; }
    public PinDirection Direction { get; set; }
    public VoltageLevel? VoltageLevel { get; set; }
    public int? ChannelNumber { get; set; }
    public Guid? ComponentTemplatePinId { get; set; }
    public Guid? IoChannelId { get; set; }
}



/// <summary> Cihazin  <c>CurrentValue</c> ve <c>ValueUpdatedAt</c> bilgileri BILEREK YOK. Canli deger SignalR kanalından akar </summary>
public class DiagramIoChannelDto : IDto
{
    public Guid Id { get; set; }
    public int ChannelNumber { get; set; }
    public PinDirection Direction { get; set; }
    public bool IsEnabled { get; set; }
    public string Name { get; set; } = null!;
}

File: Scadex.WebUI/src/models/diagram/queries/diagramDeviceDto.ts
/** Ayna: Scadex.Model/Dtos/Diagram/Queries/DiagramDeviceDto.cs — sözleşme: docs/api-contract/02-diagram-read.md */
import type { DeviceStatus } from '@/models/enums';
import type { DiagramIoChannelDto } from './diagramIoChannelDto';
import type { DiagramPinDto } from './diagramPinDto';
import type { DiagramTemplateDto } from './diagramTemplateDto';

/**
 * Canvas'ta bir React Flow node'u olarak render edilen cihaz.
 * `position` = (coordinateX, coordinateY), `draggable` = !isLocked,
 * `hidden` = !isVisible, boyut `template`'ten gelir.
 */
export interface DiagramDeviceDto {
  id: string;
  name: string;
  coordinateX: number;
  coordinateY: number;
  /** Derece. React Flow'da rotation prop'u yok — node kökünde CSS transform. */
  rotation: number;
  zIndex: number;
  isLocked: boolean;
  isVisible: boolean;
  isActive: boolean;
  componentTemplateId: string;
  /** SCADA tarafındaki kimlik; ingest bu kodla cihaz çözümler. */
  externalCode: string | null;
  /** Null = hiç telemetri alınmadı. 0 DEĞİL — 0 `DeviceStatus.Offline`'dır. */
  deviceStatusId: DeviceStatus | null;
  deviceStatusName: string | null;
  lastSeen: string | null;
  template: DiagramTemplateDto;
  pins: DiagramPinDto[];
  ioChannels: DiagramIoChannelDto[];
}
